### PR TITLE
🎉  Feature/layer config content

### PR DESCRIPTION
**Pull request by erwagasore**
- Enabled load `contents` from `layer` configuration file
- Added `info` button on the left side of `layer legend`
- Once clicked a dialog pop-up and show `markdown` or `html` contents within `.md` file

### CODE REVIEW

**Comment by wadhwamatic** [comment]
@erwagasore - can you add some instructions on how to test this feature?

When I tried to run locally, I get this error:
./src/components/MapView/Legends/layerContentPreview.tsx
  Line 2:20:  Unable to resolve path to module 'marked'  import/no-unresolved

**Comment by erwagasore** [comment]
@wadhwamatic the only step out of the ordinary is to run `yarn install` or `npm install` for new libraries `marked` and `domsanitize` to be installed from updated `package.json`

**Comment by wadhwamatic** [comment]
Ok, I was able to compile after removing node_modules and reinstalling. But I don't know how to configure the popup content. Please provide instructions!

**Comment by wadhwamatic** [comment]
I dug around and found how you implemented the content attribute in layers.json. We need the ability to link to anchors within the markdown file. I tried adding an anchor to the markdown using # to define a header. I then tried to link to that anchor in the layers.json config. 

Specifically:
- In contents.md, I added "# Multidimensional disadvantage index"
- In layers.json I added contents_path": "data/myanmar/contents.md#multidimensional-disadvantage-index" 

But this doesn't advance to the header as expected. Thoughts?

**Comment by erwagasore** [comment]
@wadhwamatic scrolling to position inside of `Dialog` component is difficult to pull of. Here are a couple of reasons:
- We are processing the `contentPath` which may include the `#name-of-id` using `markdown` parse which doesn't relate well with reactjs processes.
- After parsing we are processing the resulting `html` into react as it is thus cannot add `ref` object to help scrolling to view programmatically. This is doable but kind of too involved. Tell me if I can go through the rabbit hole to pull it off :)

**Comment by wadhwamatic** [comment]
@erwagasore - this worked fine with the initial examples you included. But I then tried to create a new markdown file with content that would be more similar to what we'd use in practice and added the content link to a couple of layers. Unfortunately, the content did not scroll to the header as expected. See this branch: https://github.com/WFP-VAM/prism-frontend/tree/layer-content-aw

**Comment by erwagasore** [comment]
@wadhwamatic I have replaced to library to use a more flexible lib with the ability to replace `DOM Node` conditionally.
I have tested with your robust `markdown` now this pr is ready to be merge. Please review

File: src/components/MapView/Legends/layerContentPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import { LayerType } from '../../../config/types';
+import { LayerDefinitions } from '../../../config/utils';
+
+const LayerContentPreview = ({ layerId, classes }: PreviewProps) => {
+  const [open, setOpen] = useState(false);
+  const [content, setContent] = useState('');
+  const contentRef = useRef<HTMLHeadingElement>(null);
+  const layer = LayerDefinitions[layerId || 'admin_boundaries'];
+  const canDisplayContent = !layer.group || layer.group.main;
+  const hasContent = layer.contentPath && layer.contentPath.length > 0;
```
Comment (Ry-DS) [inline src/components/MapView/Legends/layerContentPreview.tsx:27]: optional chaining here?

File: src/components/MapView/Legends/layerContentPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  const domId = layer.contentPath && layer.contentPath.split('#')[1];
```
Comment (Ry-DS) [inline src/components/MapView/Legends/layerContentPreview.tsx:29]: optional chaining here too

File: src/components/MapView/Legends/layerContentPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      fetch(path)
+        .then(response => response.text())
+        .then(text => setContent(text));
+    }
+  }, [hasContent, layer.contentPath]);
+
+  const transform: HTMLReactParserOptions = {
+    replace: domNode => {
+      if (
+        domNode instanceof Element &&
```
Comment (Ry-DS) [inline src/components/MapView/Legends/layerContentPreview.tsx:55]: optional chaining here too

File: src/components/MapView/Legends/layerContentPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    },
+    title: {
+      color: theme.palette.text.secondary,
+      fontWeight: 'bold',
+      minWidth: '600px',
+    },
+  });
+
+export interface PreviewProps extends WithStyles<typeof styles> {
+  layerId: LayerType['id'] | undefined;
```
Comment (Ry-DS) [inline src/components/MapView/Legends/layerContentPreview.tsx:113]: could you do `layerId?:` here?

**Review by Ry-DS — APPROVED** [review]
Very small things regarding code quality, but otherwise looks good (just assessed code quality here ;))

File: src/components/MapView/Legends/layerContentPreview.tsx (hunk 4 above)
Comment (erwagasore) [reply src/components/MapView/Legends/layerContentPreview.tsx:113]: Replaced
```typescript
export interface PreviewProps extends WithStyles<typeof styles> {
  layerId: LayerType['id'] | undefined;
}
```
with

```typescript
export interface PreviewProps extends WithStyles<typeof styles> {
  layerId?: LayerType['id'];
}
```

File: src/components/MapView/Legends/layerContentPreview.tsx (hunk 2 above)
Comment (erwagasore) [reply src/components/MapView/Legends/layerContentPreview.tsx:29]: Replaced
```typescript
const domId = layer.contentPath && layer.contentPath.split('#')[1];
```

with 

```typescript
const domId = layer.contentPath?.split('#')?.[1];
```

File: src/components/MapView/Legends/layerContentPreview.tsx (hunk 1 above)
Comment (erwagasore) [reply src/components/MapView/Legends/layerContentPreview.tsx:27]: Replaced

```typescript
const hasContent = layer.contentPath && layer.contentPath.length > 0;
```

with

```typescript
const hasContent = layer.contentPath?.length;
```

**Review by wadhwamatic — APPROVED** [review]
Working well - nice work on this

**Pull request merged**